nonce/nonce.go

```diff
@@ -65,8 +68,18 @@ func (h *int64Heap) Pop() interface{} {
 }
 
 // NewNonceService constructs a NonceService with defaults
-func NewNonceService(scope metrics.Scope, maxUsed int) (*NonceService, error) {
+func NewNonceService(scope metrics.Scope, maxUsed int, prefix string) (*NonceService, error) {
 	scope = scope.NewScope("NonceService")
+
+	if prefix != "" {
+		if len(prefix) != 4 {
+			return nil, errors.New("nonce prefix must be 4 characters")
+		}
+		if _, err := base64.RawURLEncoding.DecodeString(prefix); err != nil {
```

Comment:
It might be worth adding a quick comment here that reminds the reader that this is an RFC 8555 6.5.1 requirement. 

>  The value of the Replay-Nonce header field MUST be an octet string encoded according to the base64url encoding described in Section 2 of [RFC7515]. 

An explicit comment would also help capture the discussion from the review of this PR that decided it was necessary for client compat.

Reply:
Done.